parser/parser.go

```diff
@@ -722,12 +723,23 @@ func (p *Parser) parseSource() Command {
 	if len(srcErrors) > 0 {
 		p.errors = append(p.errors, NewError(p.peek, fmt.Sprintf("%s has %d errors", srcPath, len(srcErrors))))
 		p.nextToken()
-		return cmd
+		return []Command{cmd}
 	}
 
 	cmd.Args = p.peek.Literal
+	filtered := make([]Command, 0, len(srcCmds))
```

Comment:
we're not adding all the src into filtered, so I think probably can be 

```suggestion
	var filtered []Command
```